src/Client/Updater.php

```diff
@@ -159,48 +166,53 @@ public function refresh(bool $force = false): bool
     {
         if ($force) {
             $this->isRefreshed = false;
+            $this->metadataExpiration = null;
         }
         if ($this->isRefreshed) {
             return true;
         }
 
+        // *TUF-SPEC-v1.0.13 Section 5.0
+        $this->setMetadataExpiration();
+        $metadataExpirationTime = $this->getMetadataExpirationTime();
+
+        // *TUF-SPEC-v1.0.13 Section 5.1
         $rootData = RootMetadata::createFromJson($this->durableStorage['root.json']);
         $rootData->setIsTrusted(true);
 
         $this->roleDB = RoleDB::createFromRootMetadata($rootData);
         $this->keyDB = KeyDB::createFromRootMetadata($rootData);
 
+        // *TUF-SPEC-v1.0.13 Section 5.2
         $this->updateRoot($rootData);
 
-        $nowDate = $this->getCurrentTime();
-
-        // *TUF-SPEC-v1.0.12 Section 5.2
+        // *TUF-SPEC-v1.0.13 Section 5.3
         $newTimestampContents = $this->fetchFile('timestamp.json');
         $newTimestampData = TimestampMetadata::createFromJson($newTimestampContents);
-        // *TUF-SPEC-v1.0.12 Section 5.2.1
+        // *TUF-SPEC-v1.0.13 Section 5.3.1
         $this->checkSignatures($newTimestampData);
 
         // If the timestamp or snapshot keys were rotating then the timestamp file
         // will not exist.
         if (isset($this->durableStorage['timestamp.json'])) {
-            // *TUF-SPEC-v1.0.12 Section 5.2.2.1 and 5.2.2.2
+            // *TUF-SPEC-v1.0.13 Section 5.3.2.1 and 5.3.2.2
             $currentStateTimestampData = TimestampMetadata::createFromJson($this->durableStorage['timestamp.json']);
             static::checkRollbackAttack($currentStateTimestampData, $newTimestampData);
         }
 
-        // *TUF-SPEC-v1.0.12 Section 5.2.3
-        static::checkFreezeAttack($newTimestampData, $nowDate);
-        // TUF-SPEC-v1.0.12 Section 5.2.4: Persist timestamp metadata
+        // *TUF-SPEC-v1.0.13 Section 5.3.3
+        static::checkFreezeAttack($newTimestampData, $metadataExpirationTime);
```

Comment:
For a follow-up I think we don't need to pass in the date here. I think this was done to make it unit test able but I don't think we need to unit test non-public methods.